src/core/authorization_request/parameters.rs

```diff
@@ -169,6 +188,150 @@ impl ClientMetadata {
         tracing::warn!("the client metadata was not passed by reference or value");
         Ok(ClientMetadata(UntypedObject::default()))
     }
+
+    /// OPTIONAL. A JWKS as defined in
+    /// [RFC7591](https://openid.net/specs/openid-4-verifiable-presentations-1_0.html#RFC7591).
+    ///
+    /// It MAY contain one or more public keys, such as those used by the Wallet as an input to a
+    /// key agreement that may be used for encryption of the Authorization Response
+    /// (see [Section 7.3](https://openid.net/specs/openid-4-verifiable-presentations-1_0.html#jarm)),
+    /// or where the Wallet will require the public key of the Verifier to generate the Verifiable Presentation.
+    ///
+    /// This allows the Verifier to pass ephemeral keys specific to this Authorization Request.
+    /// Public keys included in this parameter MUST NOT be used to verify the signature of signed Authorization Requests.
+    ///
+    ///
+    /// See reference: https://openid.net/specs/openid-4-verifiable-presentations-1_0.html#section-5.1-4.2.2.1
+    ///
+    /// The jwks_uri or jwks metadata parameters can be used by clients to register their public encryption keys.
+    ///
+    /// See: https://openid.net/specs/oauth-v2-jarm-final.html#section-3-4
+    ///
+    pub fn jwks(&self) -> Option<Vec<JWK>> {
+        self.0 .0.get("jwks").and_then(|j| j.as_array()).map(|j| {
+            j.iter()
+                // NOTE: we're ignoring the errors here, but may wish to handle the errors in the future.
+                .filter_map(|jwk| serde_json::from_value(jwk.clone()).ok())
```

Comment:
This and the other parameters that you have manually implemented are already implemented here:
https://github.com/spruceid/openid4vp/blob/main/src/core/metadata/parameters/verifier.rs

You should change these methods to `self.0.get()` as for vp_formats, or just do `client_metadata.0.get()` in the downstream code.

Reply:
Thanks for pointing this out. I updated to use the typed parameter implementations, and added missing typed parameter implementations for the other fields.